wallet-new/integration/Util.hs

```diff
@@ -170,11 +174,80 @@ noLongerThan action maxWaitingTime = do
         Left _  -> return Nothing
         Right a -> return $ Just a
 
-
 -- | Output for @Text@.
 printT :: Text -> IO ()
 printT = putStrLn
 
 randomTest :: (Testable a) => String -> Int -> PropertyM IO a -> Spec
 randomTest msg maxSuccesses =
     prop msg . withMaxSuccess maxSuccesses . monadicIO
+
+log :: MonadIO m => Text -> m ()
+log = putStrLn . mappend "[TEST-LOG] "
+
+ppShowT :: Show a => a -> Text
+ppShowT = fromString . ppShow
+
+shouldReturnRight :: (HasCallStack, Show err) => IO (Either err a) -> IO a
+shouldReturnRight a = a >>= \case
+    Right x  -> return x
+    Left err -> do
+        expectationFailure ("expecting Right (..) got: " ++ show err)
+        error "unreachable"
+
+--shouldReturnRight :: (HasCallStack, Show err) => IO (Either err a) -> IO a
+shouldReturnData :: (HasCallStack, Show err) => IO (Either err  (WalletResponse a)) -> IO a
+shouldReturnData a = wrData <$> shouldReturnRight a
+
+shouldFailWith
+    :: HasCallStack
+    => IO (Either ClientError a)
+    -> ClientError
+    -> IO ()
+shouldFailWith action wantErr = action >>= \case
+    Right _  -> expectationFailure errMsg
+    Left err -> do
+        if err == wantErr
+            then return ()
+            else expectationFailure errMsg
+    where
+         errMsg = "expecting Left ("++ show wantErr ++ ")"
+
+shouldFail :: HasCallStack => IO (Either ClientError a) -> IO ()
+shouldFail action = action >>= \case
+    Right _  -> expectationFailure "expecting Left (..)"
+    Left _   -> return ()
+
+makePayment
+    :: WalletClient m
+    -> Core.Coin
+    -> (Wallet, Account)
+    -> Address
+    -> Resp m Transaction
+makePayment wc amount (sourceW, sourceA) destination = do
+    let payment = Payment
+          { pmtSource = PaymentSource
+               { psWalletId     = walId sourceW
+               , psAccountIndex = accIndex sourceA
+               }
+          , pmtDestinations = pure PaymentDistribution
+               { pdAddress = V1 destination
+               , pdAmount  = V1 amount
+               }
+          , pmtGroupingPolicy   = Nothing
+          , pmtSpendingPassword = Nothing
+          }
+    postTransaction wc payment
+
+chargeWallet
+    :: Http.WalletClient IO
+    -> Core.Coin
+    -> Http.Wallet
+    -> IO (Http.Account, Http.WalletAddress)
```

Comment:
Let's try to be consistent in the style here, at least within the same file :+1: 

`Http.WalletClient IO` -> `WalletClient m`, `Http.Wallet` -> `Wallet` etc ...

Reply:
Sorry I overlooked that `IO` -> `m` change. 
I thought you refer to the `Http.`prefix (changed that.)
I just clicked on resolving because github said there are unresolved conversations and
I though that it would unblock the PR.
No problem, if there are still important changes to make to the PR just reopen the conversation